src/middleware.js

```diff
@@ -155,3 +155,27 @@ export function withVersionHeader (handler) {
     return response
   }
 }
+
+/**
+ * This middleware throws if the Content-Length header is set and it is greater
+ * than the provided max value.
+ *
+ * @param {number} maxLength
+ * @param {import('@web3-storage/gateway-lib').Handler<import('@web3-storage/gateway-lib').Context>} handler
+ */
+export function withMaxContentLength (maxLength, handler) {
+  /**
+   * @type {import('@web3-storage/gateway-lib').Handler<import('@web3-storage/gateway-lib').Context>}
+   */
+  return async (request, env, ctx) => {
+    const response = await handler(request, env, ctx)
+    if (!response.headers.has('Content-Length')) return response
+
+    const contentLength = parseInt(response.headers.get('Content-Length') || '0')
```

Comment:
With the default value being provided, we can just collapse this to
```suggestion
    const contentLength = parseInt(response.headers.get('Content-Length') ?? '0')
```

Reply:
Hmm, I thought it returned `null` and `??` only work with `undefined`?

Reply:
oh it's nullish...

Reply:
👍 thanks!